test/OrchardCore.Commerce.Tests.UI/Tests/UserTests/UserPersistenceTests.cs

```diff
@@ -116,29 +119,55 @@ public UserPersistenceTests(ITestOutputHelper testOutputHelper)
             browser);
 
     [Theory, Chrome]
-    public Task AdminUserEditorShouldSaveWithoutFullyFilledUserAddresses(Browser browser) =>
+    public Task UserDetailsEditorShouldPersist(Browser browser) =>
         ExecuteTestAfterSetupAsync(
             async context =>
             {
-                async Task SubmitAndGoToUserAddressesAsync()
-                {
-                    await context.ClickReliablyOnSubmitAsync();
-                    context.ShouldBeSuccess();
+                await context.SignInDirectlyAsync();
+                await context.GoToRelativeUrlAsync("/user/details");
 
-                    await context.ClickReliablyOnAsync(By.XPath(
-                        "//li[contains(@class, 'list-group-item')][.//h5[contains(., 'TestUser')]]" +
-                        "//a[contains(@class, 'btn-primary') and contains(., 'Edit')]"));
-                    await context.GoToEditorTabAsync("User Addresses");
-                }
+                await context.ClickAndFillInWithRetriesAsync(
+                    By.Id("UserDetailsPart_PhoneNumber_Text"),
+                    PhoneNumber);
+                await context.ClickAndFillInWithRetriesAsync(
+                    By.Id("UserDetailsPart_VatNumber_Text"),
+                    VatNumber);
+                await context.ClickReliablyOnAsync(By.XPath("//label[contains(., 'User is a corporation')]"));
+
+                await context.ClickReliablyOnSubmitAsync();
+                context.ShouldBeSuccess("Your details have been updated.");
 
+                const string getInputsScript = @"return JSON.stringify(
+                    Array.from(document.querySelectorAll('form[action=\'/user/details\'] input, form[action=\'/user/details\'] select'))
```

Comment:
You can also use backticks as string delimiter in JS (even if it's not a template string), so you won't run out of quote types:
```suggestion
                    Array.from(document.querySelectorAll(`form[action='/user/details'] input, form[action='/user/details'] select`))
```